python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -858,6 +858,10 @@ def dispose(self) -> None:
         if self._secrets_loader:
             self._secrets_loader.dispose()
 
+        if self in DagsterInstance._TEMP_DIRS:
+            DagsterInstance._TEMP_DIRS[self].cleanup()
+            del DagsterInstance._TEMP_DIRS[self]
```

Comment:
are there thread safety issues here if multiple DagsterInstances are getting cleaned up at the same time, does this need a lock?